src/boost_histogram/_internal/hist.py

```diff
@@ -885,6 +890,30 @@ def __getitem__(self: H, index: IndexingExpr) -> H | float | Accumulator:
             logger.debug("Reduce actions are all empty, just making a copy")
             reduced = copy.copy(self._hist)
 
+        # bin re-grouping
+        if (
+            hasattr(ind, "step")
+            and ind.step is not None
+            and ind.step.groups is not None
+        ):
+            axes = [reduced.axis(i) for i in range(reduced.rank())]
+            reduced_view = reduced.view(flow=True)
+            new_axes_indices = [axes[i].edges[0]]
+            j: int = 0
+            for group in groups:
+                new_axes_indices += [axes[i].edges[j + 1 : j + group + 1][-1]]
+                j = group
+
+            variable_axis = Variable(new_axes_indices, metadata=axes[i].metadata)
+            axes[i] = variable_axis._axis
```

Comment:
This is a quick and easy way to get the C++ axes, just make a Python axis and pull the `._axis` out of it.